index.js

```diff
@@ -31,7 +31,14 @@
   var _crypto;
   if(typeof crypto !== 'undefined') {
     _crypto = crypto;
+  } else if (window.crypto) {
+    _crypto = window.crypto;
+  } else if (window.msCrypto) {
+    _crypto = window.msCrypto;
+  } else {
+    throw new Error('Non-standard crypto library');
```

Comment:
I think your comment above probably covers this, but it does look like this fix needs to be refactored or a bit or it will cause trouble with node.js usage.

Reply:
Absolutely!
I've updated this. Please take a look at the new changes.